.github/actions/setup-ccache/action.yml

```diff
@@ -0,0 +1,28 @@
+name: Setup sccache
+inputs:
+  extra-cache-key:
+    required: false
+    default: ''
+runs:
+  using: composite
+  steps:
+    - name: Get Date
+      id: get-date
+      run: echo "::set-output name=today::$(/bin/date -u '+%Y%m%d')"
+      shell: bash
+
+    - name: Fix Windows temporary directory
+      # On Windows, for some reason the default temporary directory provided to sccache
+      # may become read-only at some point. Work around by having a private tempdir.
+      id: mktemp
+      run: echo "::set-output name=tmpdir::$(cygpath -w $(mktemp -d))"
+      shell: bash
+      if: ${{ runner.os == 'Windows' }}
+
+    - name: Setup sccache
+      uses: hendrikmuhs/ccache-action@v1.2
+      with:
+        variant: sccache
```

Comment:
From browsing sccache, it seems like the only nice benefit is flexible cloud provider storage options which is not relevant here?

Reply:
Ccache has only recently gained Windows (MSVC) support . I’m not sure how stable it is but we can try. Sccache Windows support is older and my hypothesis is that this also means it’s more stable.